ttnn/cpp/ttnn/tensor/tensor.cpp

```diff
@@ -883,12 +882,29 @@ void memcpy(
     EnqueueReadBuffer(queue, src.device_buffer(), dst, blocking);
 }
 
-void memcpy(void* dst, const Tensor& src, const std::optional<std::size_t> transfer_size, bool blocking) {
-    memcpy(src.device()->command_queue(), dst, src, transfer_size, blocking);
+void memcpy(
+    CommandQueue& queue, void* dst, const Tensor& src, const size_t offset, const size_t size, const bool blocking) {
+    if (not is_device_tensor(src)) {
+        TT_THROW("memcpy: src tensor must be on device");
+    }
+
+    const char* TT_METAL_SLOW_DISPATCH_MODE = std::getenv("TT_METAL_SLOW_DISPATCH_MODE");
```

Comment:
don't want to query the env in a perf path, use rtoptions (or add there if not present)

Reply:
The `RunTimeOptions` class currently doesn't query for fast/slow dispatch. I think modifying the `RunTimeOptions` class to support querying for the dispatch type shouldn't be bundled with this PR and instead be done in a separate PR as `getenv("TT_METAL_SLOW_DISPATCH_MODE")` is used in a lot of places throughout the codebase.